private/bufpkg/buflock/lock_file.go

```diff
@@ -52,8 +54,15 @@ func readConfig(ctx context.Context, readBucket storage.ReadBucket) (_ *Config,
 			return nil, fmt.Errorf("failed to unmarshal lock file at %s: %w", V1Version, err)
 		}
 		config := &Config{}
+		var isB3DigestFound bool
 		for _, dep := range externalConfig.Deps {
 			config.Dependencies = append(config.Dependencies, DependencyForExternalConfigDependencyV1(dep))
+			if !isB3DigestFound && strings.HasPrefix(dep.Digest, "b3-") {
```

Comment:
This only checks b3, perhaps we should check for b1 as well?

Reply:
The mapper functions checks b1 as well https://github.com/bufbuild/buf/blob/main/private/bufpkg/buflock/buflock.go#L90